### ISSUE TITLE
Smart Ctrl+C

### ISSUE THREAD

**Issue by vlad-shatskyi** [railsware/upterm#1115]
On Linux, Ctrl+C should copy text if there exists selection, otherwise it should send SIGINT to the current command.

**Comment by vlad-shatskyi** [comment]
As usual, please vote.

**Comment by SRGOM** [comment]
I like your reasoning but I'd be against making them smart, is it okay to pop a message first time user presses C-c that interrupt is C-S-c? I don't want the situation where a program was interrupted because my selection was messed.

**Comment by vlad-shatskyi** [comment]
@SRGOM good point, inconsistent behavior can cause mistakes. What do you think if we always copy on Ctrl+C and interrupt on double Ctrl+C press?

**Comment by SRGOM** [comment]
That does sound a good middle path. On that note- I'm assuming you're doing this as a default behavior and your code actually reads from a config file a list of keystrokes/chords to map to actions? In which case C-c-c for interrupt sounds like a better default. If someone is "power user" enough to need C-c at it's original place, they'd be enough "power user" to edit config!

**Comment by SRGOM** [comment]
That does sound a good middle path. On that note- I'm assuming you're doing this as a default behavior and your code actually reads from a config file a list of keystrokes/chords to map to actions? In which case C-c-c for interrupt sounds like a better default. If someone is "power user" enough to need C-c at it's original place, they'd be enough "power user" to edit config!

**Comment by deckar01** [comment]
The context menu says Ctrl+C is Copy, but it doesn't work.

![screenshot_2018-01-16_15-49-30](https://user-images.githubusercontent.com/3108007/35014156-4fb75f12-fad5-11e7-8762-05bdaf5a796b.png)

---

A context sensitive Ctrl+C binding seems like a nice compromise. Ctrl+Shift+C is working great for me now, but I switch back and forth between a mac and linux so often it trips me up often. I actually use a mac keyboard on linux with Super mapped to Ctrl.

I would rather deprecate the Ctrl+C interrupt in favor of a modern UI solution like a red button that says "Stop".

![screenshot_2018-01-16_16-09-44](https://user-images.githubusercontent.com/3108007/35014839-c7553c18-fad7-11e7-9a2d-0e19b8915b64.png)
